app/src/main/java/com/wstxda/gsl/MusicSearchQuickSettings.kt

```diff
@@ -1,18 +1,34 @@
 package com.wstxda.gsl
 
 import android.annotation.SuppressLint
+import android.app.PendingIntent
 import android.content.Intent
+import android.os.Build
 import android.service.quicksettings.TileService
 
 class MusicSearchQuickSettings : TileService() {
     // android.service.quicksettings.TileService
     @SuppressLint("StartActivityAndCollapseDeprecated")
     override fun onClick() {
         super.onClick()
-        startActivityAndCollapse(
-            Intent("com.google.android.googlequicksearchbox.MUSIC_SEARCH").setFlags(
-                Intent.FLAG_ACTIVITY_NEW_TASK
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
+            /* use PendingIntent for the SDK and above */
+            val pendingIntent = PendingIntent.getActivity(
+                this,
+                0,
+                Intent("com.google.android.googlequicksearchbox.MUSIC_SEARCH").apply {
+                    flags = Intent.FLAG_ACTIVITY_NEW_TASK
+                },
+                PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
             )
```

Comment:
This method is correct, although it cannot collapse system qs panel, apparently this behavior is now managed by the activity being opened or by the system itself. you could probably try some **workarounds** but honestly I feel like it's not worth it.

I was able to try the changes and the activity is also not invoked on HyperOS A14 for some reason (stupid probably)

Reply:
> This method is correct, although it cannot collapse system qs panel, apparently this behavior is now managed by the activity being opened or by the system itself. you could probably try some **workarounds** but honestly I feel like it's not worth it.
> 
> I was able to try the changes and the activity is also not invoked on HyperOS A14 for some reason (stupid probably)

Hmm, unfortunately I don't have Android 14 to test.

Reply:
`code=START_ABORTED` apparently the system is blocking launch of the intent for security policy reason.

`BAL_BLOCK` suggests that the action was blocked by Activity Management,  which reinforces that somehow opening the intent is outside of security

`08-25 01:15:03.157  1062  2196 I ActivityTaskManager: START u0 {act=com.google.android.googlequicksearchbox.MUSIC_SEARCH flg=0x10000000 cmp=com.google.android.googlequicksearchbox/.MusicSearchGatewayInternal} with LAUNCH_MULTIPLE from uid 10031 pid -1 (BAL_BLOCK) result code=START_ABORTED`

Maybe that's why it doesn't initialize, maybe exploiting an internal API is the solution?

Reply:
Well, I just don't understand one thing, in SDK 34, what happens, what would this collapse be?

Reply:
We currently use `StartActivityAndCollapse` to get the behavior of, when tapping, launch the intent and close the system quick settings panel. This is required to display the search and display the results in the browser.

Without this behavior the entire process above is done without closing the panel.

As of sdk 34 this is no longer allowed, and apparently there are security restrictions. For this reason i kept the project with sdk 33.

Reply:
aaa, so when the tile is clicked the panel is not closed, now I understand.

Reply:
Exactly, and open intent is blocked for security reason